AcctThread.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/local/bin/python3
+
+#    Copyright (c) 2018 Ripple Labs Inc.
+#
+#    Permission to use, copy, modify, and/or distribute this software for any
+#    purpose  with  or without fee is hereby granted, provided that the above
+#    copyright notice and this permission notice appear in all copies.
+#
+#    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+#    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+#    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+
+# Walk back through an account node's history and report all transactions
+# that affect the account in reverse order (from newest to oldest).
+#
+# To capture results redirect stdout to a file.
+
+import json
+import re
+import sys
+import websocket
+
+from websocket import create_connection
+
+# Extract command line arguments.
+def extractArgs ():
+
+    # Default websocket connection if none provided.
+    connectTo = "ws://s2.ripple.com:443"
+
+    usage = 'usage: PyAcctThread <Account ID> [ws://<Server>:<port>]\n'\
+        'If <server>:<port> are omitted defaults to "{0}"'.format (connectTo)
```

Comment:
Python 3.6 introduced f strings (see https://www.python.org/dev/peps/pep-0498/). I strongly prefer them for my own scripts, but of course I don't have to worry about supporting multiple python interperters. Your call if you want to support python 3.5 or earlier.